src/components/forms/generic/AutocompleteFormComponent.tsx

```diff
@@ -0,0 +1,51 @@
+import type { AutocompleteElementProps } from 'react-hook-form-mui';
+import { AutocompleteElement } from 'react-hook-form-mui';
+import { Paper, styled } from '@mui/material';
+
+import { theme } from '../../../theme';
+
+type Props<T> = AutocompleteElementProps<
+	T,
+	any,
+	boolean | undefined,
+	boolean | undefined
+>;
+
+const StyledPaper = styled(Paper)({
+	'& .MuiAutocomplete-listbox': {
+		backgroundColor: theme.palette.background.default,
+		overflow: 'auto'
+	}
+});
+
+export const AutocompleteFormComponent = <T,>(props: Props<T>) => (
+	<AutocompleteElement
+		{...props}
+		textFieldProps={{
+			...props.textFieldProps,
+			sx: {
+				'& .MuiOutlinedInput-root': {
+					'& fieldset': {
+						borderColor: 'secondary',
+						borderWidth: 2,
+						borderRadius: 3,
+						color: 'text.primary'
+					}
+				},
+				'& .MuiInputLabel-root': {
+					color: 'text.primary'
+				}
+			}
+			// TODO: Figure out how to get any image there (it would be then moved to the specific components)
```

Comment:
Nepřišel jsem na to, jak v jednotlivých komponentách přidat obrázek závislý na vybrané hodnotě, maximálně tak přes nějaký useeffect, který by měnil obrázek, který by tam byl jako defaultní součást labelu. Pokud na to přijdeš @dacerondrej, mělo by se to pak umístit do jednotlivých komponent, které ten obrázek používají, a tady smazat.